app/models/concerns/hashed_password_overrides.rb

```diff
@@ -0,0 +1,55 @@
+# In order to perform scrypt calculation of hashed password in a single
+# place for both password and PII encryption, we override
+# a few methods to build the encrypted_password via KeyMaker.
+#
+module HashedPasswordOverrides
+  extend ActiveSupport::Concern
+
+  attr_accessor :hashed_password
+
+  # hashed_password and unlock_password are new methods.
+  def unlock_password(password)
+    self.hashed_password = PasswordHasher.hash(password, authenticatable_salt)
+    key_maker.unlock(hashed_password, encryption_key)
+    hashed_password
+  end
+
+  def self.required_fields(_klass)
+    [:password_salt, :encryption_key]
+  end
+
+  def password_digest(password)
+    hashed_password = PasswordHasher.hash(password, authenticatable_salt)
+    key_maker.make(hashed_password)
+    self.encryption_key ||= hashed_password.encryption_key
+    hashed_password.hash_f
+  end
+
+  def valid_password?(password)
+    return false if encrypted_password.blank?
+    begin
+      unlock_password(password)
+    rescue Pii::EncryptionError => _err
+      return false
+    end
+    encrypted_password == hashed_password.hash_f
```

Comment:
Is it worth using `ActiveSupport::SecurityUtils.secure_compare` to try to mitigate timing attacks?